### DISCUSSION TITLE
How can i contribute to this project

### DISCUSSION THREAD

**Discussion by abhi808shek** [Hyperfoil/Horreum#355] [General]
Hlo Mentors my self Abhishek Gupta and i am a student and i am totally new to open source can you please help me how can I contribute to this project not as a intension to be selected in gsoc but at a mindset of learning new things and contributing as many as i can contribute to the community.

**Comment by jesperpedersen** [comment]
Hi,

Contributing to an Open Source project is based on interest / passion for a certain project / area. Horreum is about providing an advanced web UI to a performance repository. The project uses many technologies in order to make that happen.

I would start with the https://github.com/Hyperfoil/Horreum/blob/master/README.md file, and get the project running. Then look at the technologies involved, and use the software. You may find something that is missing; documentation, UI, data model, ... that you want to work on fixing. We can then discuss what you have found.

We have issues recorded - some of them are easier than others. F.ex. if you are interested in web UI development then https://github.com/Hyperfoil/Horreum/issues/236 could be an idea to begin with in order to get to know the source code.

Thanks !

**Reply by abhi808shek** [reply]
I have cloned the project but i am not able to figureout how can i run the project in my localhost

**Reply by jesperpedersen** [reply]
See "To run without test cases do" in the README.md after you have the "Prerequisites" sorted

**Reply by abhi808shek** [reply]
It is taking to much time to setup its look like that it is downloading the same package again and again after running the mvn -DskipTest=true package

**Reply by jesperpedersen** [reply]
The files will be cached for the next run - take some time to learn about Maven here https://maven.apache.org/

**Reply by abhi808shek** [reply]
Before 35 min ago i am looking the same thing it is downloading the same package again and again which package takes this much of time to download

**Reply by jesperpedersen** [reply]
It will try and download the file until it is valid locally

**Reply by abhi808shek** [reply]
So approx how much time it will take to download all the files ??

**Reply by jesperpedersen** [reply]
Depends on your setup - here is takes <2 minutes

**Reply by abhi808shek** [reply]
Can i run the next command parallely in another terminal to save time ??

**Comment by abhi808shek** [comment]
![Uploading image.jpg…]()

**Reply by abhi808shek** [reply]
After taking 40 min it shown me this error

**Comment by jesperpedersen** [comment]
Did you try other Maven based projects ?

**Reply by abhi808shek** [reply]
No sir this will be the first one

**Reply by abhi808shek** [reply]
Finally i have setup frontend and run successfully now it is the time to run the backend  part

**Reply by abhi808shek** [reply]
After loading site it is showing me error that failed to retrieve a list of existing folders how can resolve this

**Reply by jesperpedersen** [reply]
https://github.com/Hyperfoil/Horreum#example-configuration ?

**Reply by abhi808shek** [reply]
I am using windows 10 how can i configure their

**Reply by abhi808shek** [reply]
The file structure you share is of linux

**Reply by jesperpedersen** [reply]
So, that would be a good contribution -- get the project running on Windows using either docker-compose or podman-compose.

Using docker for you is likely easier... But, I don't use Windows.

**Reply by abhi808shek** [reply]
Okay sir let me try docker
